mailbox/lucene/src/test/java/org/apache/james/mailbox/lucene/search/LuceneMailboxMessageSortOnSentDateTest.java

```diff
@@ -0,0 +1,218 @@
+import org.apache.james.mailbox.model.SimpleMailboxACL;
+import org.apache.james.mailbox.model.TestId;
+import org.apache.james.mailbox.model.TestMessageId;
+import org.apache.james.mailbox.store.SimpleMailboxMembership;
+import org.apache.james.mailbox.store.mail.model.Mailbox;
+import org.apache.james.mailbox.store.mail.model.Message;
+import org.apache.lucene.store.RAMDirectory;
+import org.junit.Before;
+import org.junit.Test;
+
+public class LuceneMailboxMessageSortOnSentDateTest {
+
+    public static final long LIMIT = 100L;
+    private LuceneMessageSearchIndex index;
+    
+    private SimpleMailbox mailbox = new SimpleMailbox(0);
+    private MailboxSession session;
+
+    private MessageUid uid1;
+    private MessageUid uid2;
+    private MessageId id1;
+    private MessageId id2;
+
+    private Date smallestDate;
+    private Date biggestDate;
+
+    protected boolean useLenient() {
+        return true;
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        session = new MockMailboxSession("username");
+        TestMessageId.Factory factory = new TestMessageId.Factory();
+
+        index = new LuceneMessageSearchIndex(null, new TestId.Factory(), new RAMDirectory(), true, useLenient(), factory);
+        index.setEnableSuffixMatch(true);
+
+        uid1 = MessageUid.of(1);
+        uid2 = MessageUid.of(2);
+
+        id1 = factory.generate();
+        id2 = factory.generate();
+
+        GregorianCalendar cal = new GregorianCalendar();
+
+        cal.set(2017, 0, 10);
+        smallestDate = cal.getTime();
+
+        cal.set(2017, 0, 12);
+        biggestDate = cal.getTime();
+    }
+
+    private void indexMailWithSentDateAndInternalDate(MessageId id, MessageUid uid, Date internalDate, String sentDate) throws Exception {
+        Map<String, String> header = new HashMap<String, String>();
+        if (sentDate != null) {
+            header.put("Date", sentDate);
+        }
+
+        SimpleMailboxMembership membership = new SimpleMailboxMembership(
+            id,
+            mailbox.getMailboxId(),
+            uid,
+            0,
+            internalDate,
+            20,
+            new Flags(Flag.DELETED),
+            "My Otherbody2".getBytes(),
+            header);
+        index.add(session, mailbox, membership);
+    }
+
+    @Test
+    public void sortOnSentDateShouldSortUsingSendDateIfPresent() throws Exception {
+        indexMailWithSentDateAndInternalDate(id1, uid1, biggestDate, "Date: Fri, 2 Jun 2017 10:37:47 -0400");
+        indexMailWithSentDateAndInternalDate(id2, uid2, smallestDate, "Date: Fri, 2 Jun 2017 11:37:47 -0400");
+
+        SearchQuery query = new SearchQuery();
+        query.andCriteria(SearchQuery.all());
+        query.setSorts(Arrays.asList(new SearchQuery.Sort(SortClause.SentDate, false)));
+        Iterator<MessageUid> result = index.search(session, mailbox, query);
+
+        assertThat(result).containsExactly(uid1, uid2);
+    }
```

Comment:
maybe you could construct the same test but with the date in the reverse order (and so the result in the reverse order) to prove that it works fine, and that it's not just indexing by id?